src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js

```diff
@@ -258,43 +262,59 @@ angular.module('esn.inbox.libs')
     }
 
     function updateSharedMailboxCache() {
+
       return withJmapClient(function(jmapClient) {
-        return jmapClient.mailbox_get({
+        return jmapClient.mailbox_changes({
           accountId: null,
-          ids: null
-        })
-          .then(function(mailboxes) {
-            return mailboxes.list;
-          })
-          .then(function(mailboxList) {
-            return _addSharedMailboxVisibility(_getSharedMailboxes(mailboxList));
-          })
-          .then(function(sharedMailboxList) {
-            const sharedMailboxCache = _getSharedMailboxes(inboxMailboxesCache);
-            const removedSharedFoldersIds = _getDifferenceById(sharedMailboxCache, sharedMailboxList);
+          sinceState: inboxMailboxesCache.state
+        }).then(function(changes) {
+          if (changes.newState !== inboxMailboxesCache.state) {
+            return jmapClient.mailbox_get({
+              accountId: null,
+              ids: null
+            })
+              .then(function(mailboxes) {
+                _updateMailBoxCacheState(mailboxes.state);
+
+                return mailboxes.list;
+              })
+              .then(function(mailboxList) {
+                return _addSharedMailboxVisibility(_getSharedMailboxes(mailboxList));
+              })
+              .then(function(sharedMailboxList) {
+                return _updateSharedMailboxList(sharedMailboxList);
+              })
+              .then(function() {
+                $rootScope.$broadcast(INBOX_EVENTS.SHARED_FOLDERS_UPDATED);
```

Comment:
All this part could be put in a function like `fetchMailboxesAndUpdateSharedMailboxCache`, so this would reduce complexity of the `updateSharedMailbox` function.

Reply:
I prefer separate fetch and update rather than mix things in a function in this case 
and i'm using `_updateSharedMailboxList` so the code is more clear now